### PR TITLE
Bug 1194127 - Readd l10n_date to Settings for relativeDate handling in Downloads

### PR DESCRIPTION
**Pull request by zbraniecki**
https://bugzilla.mozilla.org/show_bug.cgi?id=1194127

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1cb57842043f6c0fafa1afea0925b1e603867449

File: shared/js/download/download_formatter.js [hunk 1]
```diff
@@ -75,8 +75,7 @@
         }
 
         LazyLoader.load(['shared/js/l10n_date.js'], function onload() {
-          var prettyDate = navigator.mozL10n.DateTimeFormat().fromNow(date);
-          resolve(prettyDate);
+          navigator.mozL10n.DateTimeFormat().relativeDate(date).then(resolve);
```
Comment (stasm) [inline shared/js/download/download_formatter.js]: This makes me wish `LazyLoader` offered a Promise-based API, too.

Comment (gasolin) [inline shared/js/download/download_formatter.js]: LazyLoader does have Promise-based API, you can use `LazyLoader.load([]).then(()=> {})` ...

Comment (stasm) [inline shared/js/download/download_formatter.js]: Oh, that's great!  @zbraniecki, sounds like you could use that instead of creating a new `Promise` then.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9859cdc35cba11fa35807e7096197a85f730e1ab

File: shared/js/download/download_formatter.js (hunk 1 above)
Comment (zbraniecki) [inline shared/js/download/download_formatter.js]: great! Updated the patch.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=229a2e4c2caf880c5f102a2627414700ba965f35

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7ef5936b8b297f909cb8494e3664614b99d886d8

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d9ae7f14b3d67afdeb06e859f93bc34c8158ce23

**Pull request merged**